### ISSUE TITLE
observables with Map() breaks the API. 

### ISSUE THREAD

**Issue by seivan** [selsamman/proxily#4]
On an class that has the field: `private _todos: Map<string, Todo> = new Map()` with the instance of the class wrapped in `observable(new MyClass())`
`_todos(key, val)` should return the map itself (for chaining or what not), but I get `undefined`
<img width="726" alt="image" src="https://user-images.githubusercontent.com/55424/168994711-06da02c3-6197-4c22-beb6-6750323b379d.png">

Works if I remove the observable, so it's an issue in the proxy implementation. Mobx does make a proxy out of it, but the API remains the same.

**Comment by seivan** [comment]
Could it be that it's missing a return statement here? 
```ts 
        switch (prop) {
            case 'get':
                return (key) => propertyReferenced(target, key, targetValue.call(target, key), (proxy) => map.set(key, proxy));
            case 'set':
                return (key, newValue) => {
                    const oldValue = map.get(key);
                    newValue = propertyUpdated(target, '*', newValue, oldValue);
                    // Change the value in the target
                    targetValue.call(target, key, newValue);
                    // Notify referencing object that referenced property has changed
                    if (target.__transaction__.timePositioning)
                        target.__transaction__.recordUndoRedo(() => mapProxy.set(key, oldValue), () => mapProxy.set(key, newValue));
                    DataChanged(target, key, true, newValue);
                };
```

**Comment by selsamman** [comment]
Thanks so much for this great catch.  And especially for identifying the actual problem in the code.  Fixed in 1.1.4.
